app/services/foreman_remote_execution/renderer_methods.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+# macros to fetch information about invoked jobs
+module ForemanRemoteExecution
+  module RendererMethods
+    extend ActiveSupport::Concern
+
+    def load_job_invocations(search: '', includes: nil, preload: nil)
+      load_resource(klass: JobInvocation, search: search, permission: 'view_job_invocations', includes: includes, preload: preload)
+    end
+
+    def load_job_invocation_by_id(id, includes: nil, preload: nil)
+      search_term = "id=#{id}"
+      begin
+        # double 'first' call due to BatchEnumerator and ActiveRecord::Relation
+        load_job_invocations(search: search_term, includes: includes, preload: preload).first.first
+      rescue NoMethodError
```

Comment:
why an unknown id raises NoMethodError? is that due to first batch returns nil? if that's the case, I'd rather check for result explicitly, like

```ruby
if (batch = load_job_invocations(...))
  batch.first
else
  raise N_("...")
end
```

the reason is, this rescue block would catch also other errors, e.g. if `load_job_invocations` would be calling something that doesn't exist